### ISSUE TITLE
browser module preview zooming (for hidpi screens)

### ISSUE THREAD

**Issue by TimUtting** [VCVRack/Rack#1395]
### Summary
The module previews are a great workflow improvement on my second screen but on my main (28in 4k, 125% system scaling) the images are far too small for me to make out with any degree of efficiency. I've tried a few different dpi scaling settings in properties of the Rack shortcut but I've not found any that make the previews big enough so it'd definitely be useful for me if there was some way to change the module preview size

### Proposed design
I think a slider in the view menu much like the overall zoom, cable opacity etc would be an ideal implementation, also ctrl scrolling while hovering over the browser (with zoom centered around the module the cursor is hovering over and possibly all others adjusting to keep it under the cursor) would be a very fast way to get around the browser but probably more difficult to implement

**Comment by Coirt** [comment]
https://github.com/VCVRack/Rack/issues/1239#issuecomment-478331417

**Comment by stoermelder** [comment]
Maybe @AndrewBelt considers adding a setting in `settings-v1.json` for that.
[modelBrowserZoom.patch.txt](https://github.com/VCVRack/Rack/files/3318582/modelBrowserZoom.patch.txt)

**Comment by Coirt** [comment]
Hopefully at some stage.

**Comment by fduhem** [comment]
#1454 https://github.com/VCVRack/Rack/pull/1454

**Comment by Coirt** [comment]
@fduhem 75% is perfect on a 32" screen, once set there is no need to change. Might I suggest moving the dropdown out of the way to the left hand side of the module browser. I can imaging others will have there own preference, a slider like in view menu would keep it consistent, would that work?

**Comment by fduhem** [comment]
@AndrewBelt does not seem to be interested, I close the Pull Request.

**Comment by Coirt** [comment]
I'd say if you made it consistent with how rack zooms it would probably get a look in. A json option is all that is needed though imo. Once set there is no reason to change it.

**Comment by fduhem** [comment]
Thanks @Coirt, I added a Slider in the main menu. Do you have any other recommendations before I submit the PR again? https://github.com/fduhem/Rack/commit/4115d361044e9665fdd4e2cd882c6e1de6e0303e

**Comment by Coirt** [comment]
In the view menu, not sure it makes sense being in there if it is a slider, having to keep going back to view to set. If it was a slider on the module browser, but on the right hand side out of the way it would be better. 

Wondering though would an option in the browser to show/hide the slider in the browser make a little more sense. Reason being the option is there if someone has a hard time seeing the browser at regular zooms but for anyone just wanting a set and forget they can hide it so it does not distract. By hide I mean `Visible = !Visible:` so it still keeps the setting.

**Comment by fduhem** [comment]
Sorry for the misunderstanding. In my initial PR there was a slider in the Module Browser (https://github.com/fduhem/Rack/commit/58c1c6723c38c371e4fda54d8035682025b494c5). In use I did not find it practical and the performance was impacted, indeed when the value of the slider changed it was necessary to regenerate the previews. It was very slow on my mac. I added the drop-down menu for selecting the zoom level because I found this behavior ergonomic.

![image](https://user-images.githubusercontent.com/235542/61090282-4986a800-a43e-11e9-9d33-8b705f9e1041.png)

![image](https://user-images.githubusercontent.com/235542/61090258-3542ab00-a43e-11e9-957b-045a8eb07f5c.png)

**Comment by fduhem** [comment]
Maybe @AndrewBelt could give us his opinion so that I can do a better PR. This feature is important to properly support hidpi screens.

**Comment by Coirt** [comment]
@fduhem There is another PR I saw that keeps the module previews in memory wonder would that have better performance.
https://github.com/VCVRack/Rack/pull/1474

**Comment by jedgarpark** [comment]
Any way to increase zoom of the module browser would be greatly appreciated.

**Comment by teknico** [comment]
Indeed. I have to use the desktop's manual zooming to make sense of the module browser.

**Comment by fractalgee** [comment]
ok, had not found this one, so adding my vote here, more than 125% zoom in windows on a 4K screen makes things like Browser look like a comic strip than a usable piece of GUI, so that is my preferred zoom. So having a setting to embiggen things like tooltips and modules in browser would be super duper

**Comment by AndrewBelt** [comment]
Why don't all of you just change your OS's display scaling? What's the point of having such tiny window title bars like in @fduhem's screenshot? If the text of your OS's windows is too small to see, isn't that a more fundamental problem than the text of a certain single application?

**Comment by fractalgee** [comment]
zoom over 125% on a 4K screen is defeating having the extra screen
realestate, and OS text is fine, as it is black on white. Grey on grey is a
not very eye friendly combo, so would be nice to be able to make that
higher contrast. or simply bigger and easier to discern, as I tested and
increasing windows scaling does NOT increase tool tip sixe, and not
increase module size in browser and makes everything else comically large,
so no thanks....

**Comment by TimUtting** [comment]
If my dpi scaling was an issue in my daily computer usage I wouldn't have it at that setting. I'm not trying to troubleshoot my entire workflow, just put in a feature suggestion that seems to be a common desire for people with high dpi screens

**Comment by catronomix** [comment]
Same for me on Windows. DPI scaling is at 150% and the text is big enough. It's the module image previews that appear too small. The names on the face of most modules appear smaller than the tooltip text and are mostly unreadable without pressing my face against the screen, so with similar-looking modules (for example Audible Instruments) I have to squint or mouse-over each one until I find what I'm looking for. Especially if I don't remember or know what I'm looking for and want to make out the functions of a module visually. 

![image](https://user-images.githubusercontent.com/42138122/72299060-1547aa80-3660-11ea-8999-1957b235b7dd.png)

![image](https://user-images.githubusercontent.com/42138122/72299212-740d2400-3660-11ea-9650-0dc9f72ba925.png)

A good size would be to have 1 rail height (where usually there is some blank space on modules with the name on it) to match the tooltip text line height.

**Comment by AndrewBelt** [comment]
@catronomix Oh right, I'm not talking about Module Browser previews. I'm talking about the size of Rack's UI, such as those tooltips, relative to the OS text size.

**Comment by fractalgee** [comment]
Some nifty setting to draw module previews in browser at 125% or 110% or
whatever one wishes would be groovy ! And same with tooltext size, pretty
please, at 125% screen zoom they still seem smaller than the OS text on my
machine, likely due to being light grey on dark background..., let us make
toottext somewhat bigger (2 points bigger) would be super duper!, I dislike
screen zoom of 150%, that is excessive and negates all the extra pixels on
this hi-DPI screen (4K)

**Comment by catronomix** [comment]
> @catronomix Oh right, I'm not talking about Module Browser previews. I'm talking about the size of Rack's UI, such as those tooltips, relative to the OS text size.

Isn't the title of this issue "browser module preview zooming" ? :)

**Comment by ThePython10110** [comment]
If you install Stoermelder's PackOne, it gives you a MB module. Add that to your patch, right-click on it, and choose v1 mod. Also, I'm pretty sure that it will appear in the Extras menu as well.

Just realized that this is a year old...
